libs/auth/src/artifact_authentication.ts

```diff
@@ -164,35 +165,15 @@ function deserializeArtifactSignatureBundle(
  */
 async function validateSigningMachineCert(
   config: ArtifactAuthenticationConfig,
-  signingMachineCert: Buffer,
-  artifact: ArtifactToImport
-): Promise<void> {
-  const certDetails = await parseCert(signingMachineCert);
-  switch (artifact.type) {
-    case 'cast_vote_records': {
-      assert(
-        certDetails.component === 'central-scan' ||
-          certDetails.component === 'scan',
-        'Signing machine cert for cast vote records should be a VxCentralScan or VxScan cert'
-      );
-      break;
-    }
-    case 'election_package': {
-      assert(
-        certDetails.component === 'admin',
-        'Signing machine cert for election package should be a VxAdmin cert'
-      );
-      break;
-    }
-    /* istanbul ignore next: Compile-time check for completeness */
-    default: {
-      throwIllegalValue(artifact, 'type');
-    }
-  }
+  signingMachineCert: Buffer
+): Promise<MachineDetails> {
   await verifyFirstCertWasSignedBySecondCert(
     signingMachineCert,
     config.vxCertAuthorityCertPath
   );
+  const signingMachineDetails =
+    await parseMachineDetailsFromCert(signingMachineCert);
+  return signingMachineDetails;
```

Comment:
in the process of this change you've made an improvement – no trying to parse until the signature is verified. That is important to protect against potential malicious data formats that try to screw up our parsing.

Reply:
Ah nice! That makes sense